### DISCUSSION TITLE
Is there any way to stop the recreation of resources if a new provider is attached to them?

### DISCUSSION THREAD

**Discussion by mate201** [pulumi/pulumi#7596] [Pulumi Providers] [answered]
We recently found ourselves in need of adding a new AWS provider in place of a default one. The problem is that after the actual addition, Pulumi tries to recreate all resources with a new provider. I was trying to solve the issue, and the only hackaround I've found is to run Pulumi up specifying only a new provider. Thus Pulumi will create a new provider in a stack and update resources with a new provider without replacing them.

Recommendations:

- Run pulumi up before provider addition to update the state and not make any other change except provider later on.

- Back up you stack with `pulumi stack export`

- Add new provider in code and run `pulumi up --target <urn_of your_new_provider>` e.g., ` pulumi up --target urn:pulumi:mate201-test-cluster-uat::uat::pulumi:providers:aws::commonAwsProvider`

**Comment by nesl247** [comment, marked as answer]
I've had this issue for a long time, and would love to see this officially solved, as pulumi should be smarter about how it handle's provider changes. If you haven't already, I would open an issue from this discussion as I think it's a real problem.

**Reply by katcosgrove** [reply]
Closing this discussion, as I've moved it to an issue [here](https://github.com/pulumi/pulumi/issues/8959)!
